### ISSUE TITLE
Updating broken web mentions

### ISSUE THREAD

**Issue by billbennettnz** [snarfed/bridgy#907]
This might be something you could add to the FAQ.

I have a couple of Twitter webmentions on my WordPress site where the original poster has changed their Twitter handle. These appear as broken links. In at least one case, I know the replacement Twitter handle. I'm told it's possible to update web mentions, but the fields are greyed out on the Wordpress comment edit page. How can I update the web mention?

**Comment by dshanske** [comment]
Editing the links manually can be done... I referred because I thought there might be something to track this at the Bridgy level.

**Comment by dshanske** [comment]
The links should be editable ob the comment edit fields, which field is greyed out?

**Comment by billbennettnz** [comment]
Probably easier to show it:

**Comment by billbennettnz** [comment]
You probably know this, but it may be useful for others later. When you scroll back through bridgy webmentions the URL includes ?responses_before=2019-05-27T21:09:05.926421#

All I had to do was find the date stamp on the original comment, tinker with the URL… and bingo. 

Refreshing the webmention didn’t fix the link though…

**Comment by snarfed** [comment]
thanks @billbennettnz! good idea.

I'll keep this issue open here to track the larger idea of tracking all sent webmentions and resending when they change.

I'm also surprised that retrying didn't work. could you post the bridgy webmention source URL? or a link to your user page with the right ?responses_before= param? I'm vacationing without a computer right now, but I'd like to look when I'm back.

**Comment by Ryuno-Ki** [comment]
Hm, assuming you are (at least temporarily - like for a couple of months) storing webmentions in a database … a form to search it would be handy.
Above, we've seen the filter for datetime.

I'd assume, @billbennettnz would have the link to the webmention he was asked to update, so being able to search for that one, would help here as well.

What other fields could be generally of interest?

**Comment by snarfed** [comment]
@Ryuno-Ki yes! agreed. luckily, bridgy already supports that! you can plug a silo or original post URL into the _ Resend for post:_ text box on your user page, and it will resend all the webmentions it knows about.

<img width="575" alt="image" src="https://user-images.githubusercontent.com/778068/76448031-f119f880-6386-11ea-9f0d-d8e9cb19ea7f.png">

this issue is more involved though. it's about updating replies (etc) from people who have _changed their username_. that's complicated for a handful of reasons beyond just finding and resending past webmentions. (fortunately, username changes are relatively rare!)

**Comment by snarfed** [comment]
For the record, Bridgy does currently detect and resend _recent_ webmentions when they change. For example, if someone comments on a GitHub issue, and then edits their comment, Bridgy would notice the change and resend the comment. 

("Recent" here isn't based on time as much as post velocity. Bridgy will generally see changes to responses on the last 10-50 or so of the user's posts.)
